org.testeditor.tcl.dsl/src/main/java/org/testeditor/tcl/dsl/jvmmodel/CallTreeBuilder.xtend

```diff
@@ -5,41 +5,64 @@ import javax.inject.Singleton
 import org.eclipse.emf.ecore.EObject
 import org.testeditor.tcl.AbstractTestStep
 import org.testeditor.tcl.CallTreeNode
+import org.testeditor.tcl.Macro
 import org.testeditor.tcl.StepContainer
 import org.testeditor.tcl.TestCase
 import org.testeditor.tcl.TestStep
 import org.testeditor.tcl.TestStepContext
 import org.testeditor.tcl.dsl.messages.TclElementStringifier
 import org.testeditor.tcl.impl.TclFactoryImpl
 import org.testeditor.tcl.util.TclModelUtil
-import org.testeditor.tcl.Macro
 
 @Singleton
 class CallTreeBuilder {
 
-	static val testSetupDisplayName = 'Setup'
-	static val testCleanupDisplayName = 'Cleanup'
+	static val testSetupDisplayName = 'setup'
+	static val testCleanupDisplayName = 'cleanup'
+	static val testConfigName = 'Config'
+	static val testLocalName = 'Local'
+
+	var String idPrefix = TclJvmModelInferrer.ID_PREFIX_TEST
 
 	@Inject extension TclFactoryImpl tclFactory
 	@Inject extension TclModelUtil
 	@Inject extension TclElementStringifier
 
 	def CallTreeNode buildCallTree(TestCase model) {
 		return model.namedCallTreeNode => [
-			val setups = (model.config?.setup ?: #[]) + model.setup
-			val cleanups = model.cleanup + (model.config?.cleanup ?: #[])
+			val parentSetup = (model.config?.setup ?: #[])
+			val parentCleanup = (model.config?.cleanup ?: #[])
 
-			if (!setups.empty) {
-				children += callTreeNodeNamed(testSetupDisplayName) => [
-					children += setups.flatMap[toCallTreeChildren]
+			treeId = 'IDROOT'
+
+			runningNumber = 0
+			idPrefix = TclJvmModelInferrer.ID_PREFIX_CONFIG_SETUP
+			if (!parentSetup.empty) {
+				children += callTreeNodeNamed(#[testConfigName, testSetupDisplayName].join(' ')) => [
```

Comment:
Isn't this overkill just for concatenating two strings, separated by a single space…? Unless I am missing something, I would suggest a more idiomatic approach, e.g.
```suggestion
				children += callTreeNodeNamed('''«testConfigName» «testSetupDisplayName»''')) => [
```
or simply
```suggestion
				children += callTreeNodeNamed(testConfigName + ' ' + testSetupDisplayName)) => [
```